tools/proofers/images_index.php

```diff
@@ -110,7 +111,7 @@ function list_images($project, $image_names, $these_are_page_images)
     echo "<h4 class='center-align'>$header</h4>";
 
     $show_replace_links = $project->can_be_managed_by_current_user;
-    $show_delete_links = $project->can_be_managed_by_current_user && !$these_are_page_images && ($project->state == PROJ_NEW || $project->state == PROJ_P1_UNAVAILABLE);
+    $show_delete_links = $project->can_be_managed_by_current_user && !$these_are_page_images && ($project->state == PROJ_NEW || $project->state == PROJ_P1_UNAVAILABLE || user_is_a_sitemanager());
```

Comment:
These conditionals are getting a little complex. I wonder if it makes sense to break them up some? Or perhaps extract them to a function so we can unit tests the truth table?

Reply:
This could be a function of the Project class, something like:
```php
public function can_user_delete_images($image_type = null)
{
    // SAs can always delete images, page or illustrations
    if (user_is_a_sitemanager()) {
        return true;
    }

    // all other actions can only be done by users who can manage the project
    if (!$this->can_be_managed_by_current_user) {
        return false;
    }

    // if page isn't an illustration but the project is New that's ok
    if ($page_type == "page" && ..
}
```

That just breaks apart this one-line into a readable function.

Reply:
This PR is only for illustration images, so, I'm guessing the function name should be `can_user_delete_nonpage_images` or something like that? SAs can already delete proofing images from the page details page at any point. If a page image is deleted, the text should be deleted, too, and I don't think we want to deal with that from the images index, so unless you think this should be a multi-purpose function, I wouldn't mention page images at all.

As I thought back after I posted the above, I realized that I did ponder breaking it up into an `if` block, but ultimately opted to keep the compact form.

I like the idea of a function, but my experience with sticking functions into classes and then invoking them is zero, so, I need help with that, too. (I don't mind -- it's a good learning experience, but the learning curve must be traversed.)

Reply:
> I like the idea of a function, but my experience with sticking functions into classes and then invoking them is zero, so, I need help with that, too. (I don't mind -- it's a good learning experience, but the learning curve must be traversed.)

You can model it off of the [can_be_managed_by_user()](https://github.com/DistributedProofreaders/dproofreaders/blob/master/pinc/Project.inc#L956) function in the Project class. It's just like a regular function and you already have a Project object where you're calling it.

Reply:
*Finally* updated. The sandbox has been updated. I'm happy to do further changes -- this was new to me, so I may not have gotten everything done that I need to. My testing comes back clean, but good to have other testers.